mobile/src/main/java/org/openhab/habdroid/ui/activity/ContentController.java

```diff
@@ -605,6 +619,90 @@ public void onClick(View view) {
         }
     }
 
+    public static class HabpanelFragment extends Fragment implements
+            ConnectionFactory.UpdateListener {
+        Connection mConnection;
+
+        public static HabpanelFragment newInstance() {
+            return new HabpanelFragment();
+        }
+
+        @Override
+        public View onCreateView(@NonNull LayoutInflater inflater,
+                ViewGroup container, Bundle savedInstanceState) {
+            View view = inflater.inflate(R.layout.fragment_habpanel, container, false);
+
+            TextView retryButton = view.findViewById(R.id.retry_button);
+            retryButton.setOnClickListener(v -> loadHabpanel(view));
+            loadHabpanel(view);
+
+            return view;
+        }
+
+        private void loadHabpanel(View view) {
+            updateViewVisibility(false, true, view);
+            try {
+                mConnection = ConnectionFactory.getUsableConnection();
+            } catch (ConnectionException e) {
+                updateViewVisibility(true, false, view);
+                return;
+            }
+
+            if (mConnection == null) {
+                updateViewVisibility(true, false, view);
+                return;
+            } else {
+                updateViewVisibility(false, true, view);
+            }
+
+            HttpUrl url = mConnection.getAsyncHttpClient().buildUrl("/habpanel/index.html");
+
+            WebView webView = view.findViewById(R.id.webview);
+
+            webView.setWebViewClient(new AnchorWebViewClient(url.toString(),
+                    mConnection.getUsername(), mConnection.getPassword()) {
+                @Override
+                public void onPageFinished(WebView view, String url) {
+                    updateViewVisibility(false, false, getView());
+                }
+
+                @Override
+                public void onReceivedError(WebView view, WebResourceRequest request,
+                        WebResourceError error) {
+                    updateViewVisibility(true, false, getView());
+                }
```

Comment:
From https://developer.android.com/reference/android/webkit/WebViewClient.html#onReceivedError(android.webkit.WebView,%20android.webkit.WebResourceRequest,%20android.webkit.WebResourceError):
```
Note that unlike the deprecated version of the callback, the new version will be called for any resource (iframe, image, etc.), not just for the main page.
```
So we probably don't want to do this unconditionally?

Reply:
I didn't see that. I need to find a better way to hide the webview if there are connection errors.

Reply:
Could check `request.getUrl()`?